packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,192 @@
+import SelectorEngine from './dom/SelectorEngine';
+import BaseComponent from './BaseComponent';
+import { enableToggleAutoloader } from './utils';
+
+const NAME = 'fileUploader';
+const IS_DRAGGABLE_CLASS_NAME = 'has-drag-and-drop';
+const IS_DRAGGING_CLASS_NAME = 'is-dragging';
+const FILE_SIZE_LIMIT = 10000000; // = 10mb
+
+class FileUploader extends BaseComponent {
+  element: HTMLElement;
+  wrapper: HTMLElement;
+  inputElement: HTMLInputElement;
+  listElement: HTMLElement | null;
+  dropZone: HTMLElement | null;
+  isDraggable: boolean;
+  isMultiple: boolean;
+  isDragging: boolean;
+  fileQueue: File[];
+
+  constructor(element: HTMLElement) {
+    super(element);
+
+    this.element = element;
+    this.wrapper = SelectorEngine.findOne('.FileUploaderInput', this.element) as HTMLInputElement;
+    this.inputElement = SelectorEngine.findOne('.FileUploaderInput__input', this.element) as HTMLInputElement;
+    this.listElement = SelectorEngine.findOne('.FileUploaderList', this.element);
+    this.dropZone = SelectorEngine.findOne('.FileUploaderInput__dropZone', this.element);
+    this.isDraggable = 'draggable' in document.createElement('span');
+    this.isMultiple = this.inputElement.multiple;
+    this.isDragging = false;
+    this.fileQueue = [];
+
+    this.init();
+    this.getFileQueue.bind(this);
+    this.clearFileQueue.bind(this);
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  public getFileQueue() {
+    return this.fileQueue;
+  }
+
+  public clearFileQueue() {
+    this.fileQueue = [];
+    this.listElement!.textContent = '';
+  }
+
+  appendToList(file: File) {
+    if (!this.listElement) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Unfortunately, there is no list element created for inserting files to upload ⚠️');
+      }
+
+      return;
+    }
+
+    if (!this.isMultiple) {
+      this.clearFileQueue();
+    }
+
+    // Creating an element to insert into a list
+    const id = file.name.replace(/\.[^/.]+$/, '');
+    const listElementNode = document.createElement('li');
+    listElementNode.setAttribute('id', id);
+    listElementNode.classList.add('FileUploaderAttachment');
+    listElementNode.innerHTML = `
```

Comment:
I think it will be better if this template will be a constant in which you will replace the content, e. g.:
```js
const template = `...`;

createUploadedFileElement({ filename, dismissableId })
// takes template
// replace special characters in the template with content
// creates a file element node
// returns the node

Then

const uploadedFileElement = this.createUploadedFileElement(...);
this.listElement.appendChild(uploadedFileElement);

Or it can be wrapped in another method that will create the list node and appends this content.
```

Reply:
I already have it ready. After yesterday's debate with @adamkudrna